docs/content/guides/dagster/branch_depl_with_snowflake.mdx

```diff
@@ -0,0 +1,325 @@
+
+Dagster automatically sets certain environment variables containing deployment metadata, allowing us to read these environment variables to discern between deployments. For a full list of preset environment variables, visit this link. We can access the `DAGSTER_CLOUD_IS_BRANCH_DEPLOYMENT` environment variable to determine the currently executing environment.
+
+Because we want to configure our assets to write to Snowflake using a different set of credentials and database in each environment, we’ll configure a separate IO manager for each environment:
+
+```python file=/guides/dagster/local_to_prod_tutorial/branch_depl_tutorial/repository_v1.py startafter=start_repository endbefore=end_repository
+@repository
+def repo():
+    snowflake_config = {
+        "account": {"env": "SNOWFLAKE_ACCOUNT"},
+        "user": {"env": "SNOWFLAKE_USER"},
+        "password": {"env": "SNOWFLAKE_PASSWORD"},
+        "schema": {"env": "SNOWFLAKE_SCHEMA"},
+    }
+    resource_defs = {
+        "branch": {
+            "hn_client": hn_api_client,
+            "snowflake_io_mgr": snowflake_io_manager.configured(
+                {
+                    **snowflake_config,
+                    "database": f"PRODUCTION_CLONE_{os.getenv('DAGSTER_CLOUD_PULL_REQUEST_ID')}",
+                }
+            ),
+        },
+        "production": {
+            "hn_client": hn_api_client,
+            "snowflake_io_mgr": snowflake_io_manager.configured(
+                {
+                    **snowflake_config,
+                    "database": "PRODUCTION",
+                }
+            ),
+        },
+    }
+
+    def get_current_env():
+        is_branch_depl = os.getenv("DAGSTER_CLOUD_IS_BRANCH_DEPLOYMENT")
+        assert is_branch_depl != None  # env var must be set
+        return "branch" if is_branch_depl else "prod"
+
+    return [
+        with_resources([items, comments, stories], resource_defs=resource_defs[get_current_env()]),
+    ]
+```
+
+### Step 3: Cloning production database per branch deployment
+
+The `branch_deployments.yml` workflow file from the [quickstart template](https://github.com/dagster-io/dagster-cloud-branch-deployments-quickstart) defines a `dagster_cloud_build_push` job that builds each branch deployment. We can modify this workflow file to add a custom step that launches a job within our repository upon deployment.
+
+We’ll first need to define a job that clones our `PRODUCTION` database for each branch deployment. Each clone will be named `PRODUCTION_CLONE_<ID>` with `<ID>` representing the pull request ID, ensuring each branch deployment has a unique clone. This job will drop a database clone if it exists, and then reclone from production, ensuring each redeployment has a fresh clone of `PRODUCTION`:
```

Comment:
idk if this is something we can add in somewhere, but i think it's pretty cool that we're using dagster to set up dbs so that we can test other dagster things. it's a neat display of a broad variety of uses for dagster